gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,593 @@
+            options.put("filter", argEntry[1].split(" "));
+            break;
+          case "pageToken":
+            // support pageToken when Cloud Resource Manager supports this (#421)
+            break;
+          case "pageSize":
+            // support pageSize when Cloud Resource Manager supports this (#421)
+            break;
+        }
+      }
+    }
+    return options;
+  }
+
+  private static final String checkForProjectErrors(Project project) {
+    if (project.getProjectId() == null) {
+      return "Project ID cannot be empty.";
+    }
+    if (!isValidIdOrLabel(project.getProjectId(), 6, 30)) {
+      return "Project " + project.getProjectId() + " has an invalid ID."
+          + " See https://cloud.google.com/resource-manager/reference/rest/" + VERSION + "/projects"
+          + " for more information.";
+    }
+    if (project.getName() != null) {
+      for (char c : project.getName().toCharArray()) {
+        if (!PERMISSIBLE_PROJECT_NAME_PUNCTUATION.contains(c) && !Character.isLetterOrDigit(c)) {
+          return "Project " + project.getProjectId() + " has an invalid name."
+              + " See https://cloud.google.com/resource-manager/reference/rest/" + VERSION
+              + "/projects for more information.";
+        }
+      }
+    }
+    if (project.getLabels() != null) {
+      if (project.getLabels().size() > 256) {
+        return "Project " + project.getProjectId() + " exceeds the limit of 256 labels.";
+      }
+      for (Map.Entry<String, String> entry : project.getLabels().entrySet()) {
+        if (!isValidIdOrLabel(entry.getKey(), 1, 63)
+            || !isValidIdOrLabel(entry.getValue(), 0, 63)) {
+          return "Project " + project.getProjectId() + " has an invalid label entry."
+              + " See https://cloud.google.com/resource-manager/reference/rest/" + VERSION
+              + "/projects for more information.";
+        }
+      }
+    }
+    return null;
+  }
+
+  private static final boolean isValidIdOrLabel(String value, int minLength, int maxLength) {
+    for (char c : value.toCharArray()) {
+      if (c != '-' && !Character.isDigit(c)
+          && (!Character.isLetter(c) || !Character.isLowerCase(c))) {
+        return false;
+      }
+    }
+    if (value.length() > 0 && (!Character.isLetter(value.charAt(0)) || value.endsWith("-"))) {
+      return false;
+    }
+    return value.length() >= minLength && value.length() <= maxLength;
+  }
+
+  Response create(Project project) {
+    project.setLifecycleState("ACTIVE");
+    project.setProjectNumber(Math.abs(PROJECT_NUMBER_GENERATOR.nextLong() % Long.MAX_VALUE));
+    project.setCreateTime(ISODateTimeFormat.dateTime().print(System.currentTimeMillis()));
+    Response response;
+    String customErrorMessage = checkForProjectErrors(project);
+    if (customErrorMessage != null) {
+      response = Error.INVALID_ARGUMENT.response(customErrorMessage);
+    } else if (projects.containsKey(project.getProjectId())) {
```

Comment:
this check and the put should be synchronized or better use `putIfAbsent` and complain if return value is not `null`.